### DISCUSSION TITLE
Which is the correct syntax to pass constructor arguments?

### DISCUSSION THREAD

**Discussion by renyuneyun** [LinkedSoftwareDependencies/Components.js#80] [Q&A] [answered]
I'm reading the [components.js document](https://componentsjs.readthedocs.io/en/latest/) recently, and found some inconsistency in its descriptions. I'm not sure whether they are both accepted, or is because of some legacy syntax.

The most noticeable question is: there are two ways of passing arguments to constructor parameters, and I'm confused about which one is the correct one.
(I'm reading mainly the typescript parts.)

In [this section in Introduction](https://componentsjs.readthedocs.io/en/latest/#3-create-a-configuration-file-to-instantiate-our-class), it says `name` is the name for a parameter in the constructor, and using `"name": "John"
` passes `"John"` to the `name` parameter.

On the other hand, in [this section in Workflow Typescript](https://componentsjs.readthedocs.io/en/latest/getting_started/basics/workflow_ts/#2-component-configuration-file), it says `"MyComponent:_name": "John"
` is the way to supply argument `"John"` to the `name` parameter.

This makes me quite confused. Are they both accepted? Or, is one of them an old way while the other is more preferred?

To me, the first one is more intuitive. But because there is no specific context for the class, it creates confusion for multiple classes. The second one is the reverse, though it also does not have context for the class.

**Comment by rubensworks** [comment, marked as answer]
> On the other hand, in [this section in Workflow Typescript](https://componentsjs.readthedocs.io/en/latest/getting_started/basics/workflow_ts/#2-component-configuration-file), it says "MyComponent:_name": "John" is the way to supply argument "John" to the name parameter.

Hmm, that one is in fact incorrect. It is missing the context of my-package. (feel free to submit a PR to fix this)

However, once that context is present, both `"name"` and `"MyComponent:_name"` are in fact accepted.
This is because using this JSON-LD context, they both expand to the same property URL.

The availability of the shorter `"name"` was added only very recently, so that's why some places may still be using the old (but still valid) `"MyComponent:_name"` version. (if you find any, PRs are welcome! :-))

**Reply by renyuneyun** [reply]
Thanks very much!
Pointing out the missing context for `my-package` gives me some more confusion now. I'll read more about that. (And submit PRs if they are still there after learning more.)
